### PR TITLE
Stack cleanups

### PR DESCRIPTION
**Pull request by AngheloAlf**
- Introduces the `STACK` and `STACK_TOP` macros.
  - Ensures the stack to always be doubleword aligned
- Fix some fake symbol usage

### CODE REVIEW

File: include/variables.h [hunk 1]
```diff
@@ -2489,11 +2485,12 @@ extern Struct_80140E80* D_801F6D4C;
 extern BombersNotebook sBombersNotebook;
 extern u8 sBombersNotebookOpen;
 extern u8 sMotionBlurStatus;
+
 extern SlowlyTask D_801F6E00;
 extern s32 D_801F6FC0;
 extern StackEntry slowlyStackEntry;
```
Comment (hensldm) [inline include/variables.h:2491]:
```suggestion
extern StackEntry sSlowlyThreadInfo;
```
To sync with other names. (This seems to be the name by the ones that are actually documented in OoT).

File: include/variables.h [hunk 2]
```diff
@@ -2489,11 +2485,12 @@ extern Struct_80140E80* D_801F6D4C;
...
-extern u8 slowlyStack[0x1000];
-extern UNK_TYPE1 D_801F7FE8;
+extern STACK(slowlyStack, 0x1000);
```
Comment (hensldm) [inline include/variables.h:2492]:
```suggestion
extern STACK(sSlowlyStack, 0x1000);
```

File: include/variables.h [hunk 3]
```diff
@@ -2512,10 +2509,10 @@ extern IrqMgrClient irqClient;
...
 extern OSMesg irqMgrMsgBuf[60];
 extern OSThread sGraphThread;
-extern u8 sGraphStack[6144];
-extern u8 schedStack[1536];
-extern u8 audioStack[2048];
-extern u8 padmgrStack[1280];
+extern STACK(sGraphStack, 0x1800);
+extern STACK(schedStack, 0x600);
+extern STACK(audioStack, 0x800);
+extern STACK(padmgrStack, 0x500);
```
Comment (hensldm) [inline include/variables.h:2515]:
```suggestion
extern STACK(sSchedStack, 0x600);
extern STACK(sAudioStack, 0x800);
extern STACK(sPadMgrStack, 0x500);
```

File: include/variables.h [hunk 4]
```diff
@@ -2512,10 +2509,10 @@ extern IrqMgrClient irqClient;
...
 extern StackEntry sGraphStackInfo;
 extern StackEntry schedStackEntry;
 extern StackEntry audioStackEntry;
```
Comment (hensldm) [inline include/variables.h:2518]:
```suggestion
extern StackEntry sSchedInfo;
extern StackEntry sAudioInfo;
```
Also `padmgrStackEntry` -> `sPadMgrInfo`

File: src/libultra/io/pimgr.c [hunk 5]
```diff
@@ -3,7 +3,7 @@
 OSPiHandle D_8009D130;
 OSPiHandle D_8009D1A8;
 OSThread D_8009D220;
-u8 piManagerStack[0x1000];
+STACK(piManagerStack, 0x1000);
```
Comment (hensldm) [inline src/libultra/io/pimgr.c:6]:
```suggestion
STACK(sPiMgrStack, 0x1000);
```

File: src/libultra/io/pimgr.c [hunk 6]
```diff
@@ -3,7 +3,7 @@
 OSPiHandle D_8009D130;
 OSPiHandle D_8009D1A8;
 OSThread D_8009D220;
```
Comment (hensldm) [inline src/libultra/io/pimgr.c:5]:
```suggestion
OSThread sPiMgrThread;
```

File: src/libultra/io/pimgr.c [hunk 7]
```diff
@@ -33,7 +33,7 @@ void osCreatePiManager(OSPri pri, OSMesgQueue* cmdQ, OSMesg* cmdBuf, s32 cmdMsgC
         __osPiDevMgr.acsQueue = &__osPiAccessQueue;
         __osPiDevMgr.piDmaCallback = __osPiRawStartDma;
         __osPiDevMgr.epiDmaCallback = __osEPiRawStartDma;
-        osCreateThread(&D_8009D220, 0, __osDevMgrMain, (void*)&__osPiDevMgr, &piManagerStack[4096], pri);
+        osCreateThread(&D_8009D220, 0, __osDevMgrMain, (void*)&__osPiDevMgr, STACK_TOP(piManagerStack), pri);
```
Comment (hensldm) [inline src/libultra/io/pimgr.c:36]:
```suggestion
        osCreateThread(&D_8009D220, 0, __osDevMgrMain, &__osPiDevMgr, STACK_TOP(piManagerStack), pri);
```

File: src/code/main.c [hunk 8]
```diff
@@ -33,24 +33,23 @@ void Main(void* arg) {
...
+    Sched_Init(&gSchedContext, STACK_TOP(sSchedStack), Z_PRIORITY_SCHED, D_8009B290, 1, &gIrqMgr);
 
     CIC6105_AddRomInfoFaultPage();
 
     IrqMgr_AddClient(&gIrqMgr, &irqClient, &irqMgrMsgQ);
 
-    StackCheck_Init(&audioStackEntry, audioStack, audioStack + sizeof(audioStack), 0, 0x100, "audio");
-    AudioMgr_Init(&audioContext, audioStack + sizeof(audioStack), Z_PRIORITY_AUDIOMGR, 0xA, &gSchedContext, &gIrqMgr);
+    StackCheck_Init(&sAudioInfo, sAudioStack, STACK_TOP(sAudioStack), 0, 0x100, "audio");
+    AudioMgr_Init(&audioContext, STACK_TOP(sAudioStack), Z_PRIORITY_AUDIOMGR, 0xA, &gSchedContext, &gIrqMgr);
```
Comment (engineer124) [inline src/code/main.c:44]:
```suggestion
    StackCheck_Init(&sAudioInfo, sAudioStack, STACK_TOP(sAudioStack), 0, 0x100, "audio");
    AudioMgr_Init(&sAudioMgr, STACK_TOP(sAudioStack), Z_PRIORITY_AUDIOMGR, 0xA, &gSchedContext, &gIrqMgr);
```
while you're here

File: src/boot_O2_g3/idle.c [hunk 9]
```diff
@@ -108,9 +108,8 @@ void Idle_InitVideo(void) {
 void Idle_ThreadEntry(void* arg) {
     Idle_InitVideo();
     osCreatePiManager(150, &gPiMgrCmdQ, sPiMgrCmdBuff, ARRAY_COUNT(sPiMgrCmdBuff));
-    StackCheck_Init(&sMainStackInfo, sMainStack, sMainStack + sizeof(sMainStack), 0, 0x400, "main");
-    osCreateThread(&gMainThread, Z_THREAD_ID_MAIN, Main_ThreadEntry, arg, sMainStack + sizeof(sMainStack),
-                   Z_PRIORITY_MAIN);
+    StackCheck_Init(&sMainStackInfo, sMainStack, STACK_TOP(sMainStack), 0, 0x400, "main");
+    osCreateThread(&gMainThread, Z_THREAD_ID_MAIN, Main_ThreadEntry, arg, STACK_TOP(sMainStack), Z_PRIORITY_MAIN);
```
Comment (engineer124) [inline src/boot_O2_g3/idle.c:112]:
```suggestion
    StackCheck_Init(&sMainStackInfo, sMainStack, STACK_TOP(sMainStack), 0, 0x400, "main");
    osCreateThread(&sMainThread, Z_THREAD_ID_MAIN, Main_ThreadEntry, arg, STACK_TOP(sMainStack), Z_PRIORITY_MAIN);
```
This looks local?

**Review by hensldm — APPROVED** [review]
1 more small nit

File: src/boot_O2_g3/boot_main.c [hunk 10]
```diff
@@ -3,19 +3,19 @@
...
-    StackCheck_Init(&sBootThreadInfo, sBootThreadStack, sBootThreadStack + sizeof(sBootThreadStack), 0, -1, "boot");
+    StackCheck_Init(&sBootThreadInfo, sBootThreadStack, STACK_TOP(sBootThreadStack), 0, -1, "boot");
     osMemSize = osGetMemSize();
     func_800818F4();
     osInitialize();
     osUnmapTLBAll();
     gCartHandle = osCartRomInit();
-    StackCheck_Init(&sIdleThreadInfo, sIdleThreadStack, sIdleThreadStack + sizeof(sIdleThreadStack), 0, 256, "idle");
-    osCreateThread(&sIdleThread, Z_THREAD_ID_IDLE, Idle_ThreadEntry, NULL, sIdleThreadStack + sizeof(sIdleThreadStack),
+    StackCheck_Init(&sIdleThreadInfo, sIdleThreadStack, STACK_TOP(sIdleThreadStack), 0, 256, "idle");
```
Comment (hensldm) [inline src/boot_O2_g3/boot_main.c:17]:
```suggestion
    StackCheck_Init(&sIdleThreadInfo, sIdleThreadStack, STACK_TOP(sIdleThreadStack), 0, 0x100, "idle");
```
Every other one uses hex.

File: include/variables.h [hunk 11]
```diff
@@ -316,7 +312,7 @@ extern FaultThreadStruct* sFaultContext;
 extern f32 D_8009BE54;
 extern u32 faultCustomOptions;
 extern u32 faultCopyToLog;
-extern u8 sFaultStack[0x600];
+extern STACK(sFaultStack, 0x600);
 extern StackEntry sFaultThreadInfo;
```
Comment (hensldm) [inline include/variables.h:316]:
```suggestion
extern StackEntry sFaultStackInfo;
```
?

File: include/variables.h [hunk 12]
```diff
@@ -2512,15 +2506,15 @@ extern IrqMgrClient irqClient;
...
+extern STACK(sAudioStack, 0x800);
+extern STACK(sPadMgrStack, 0x500);
 extern StackEntry sGraphStackInfo;
-extern StackEntry schedStackEntry;
-extern StackEntry audioStackEntry;
-extern StackEntry padmgrStackEntry;
-extern AudioMgr audioContext;
+extern StackEntry sSchedStackInfo;
+extern StackEntry sAudioStackInfo;
+extern StackEntry sPadMgrInfo;
```
Comment (hensldm) [inline include/variables.h:2516]:
```suggestion
extern StackEntry sPadMgrStackInfo;
```

File: include/variables.h [hunk 13]
```diff
@@ -2512,15 +2506,15 @@ extern IrqMgrClient irqClient;
...
+extern AudioMgr sAudioMgr;
 extern PadMgr D_801FB620;
```
Comment (hensldm) [inline include/variables.h:2518]:
```suggestion
extern PadMgr sPadMgr;
```
?

**Review by engineer124 — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]

**Pull request merged**